### PR TITLE
Feat/fly 2541

### PR DESCRIPTION
**Pull request by Hakob23**
## What

Freeze the commit-first peg-out ABI so downstream lanes can generate bindings:

- Add `IPegOutEscrow` (`requestPegOut`, `cancelPegOut`, `claimPegOut`, `getPegOutState`, refund path, and the event set).
- Extend `IFlyoverConfigurations` with `PegOutConfiguration`, `getPegOutConfiguration`, `calculatePegOutFee`, confirmation tiers, claim/fulfillment deadlines, and `maxMinerFee`.
- NatSpec on `IPegOutEscrow` records the two frozen decisions: request-id preimage (nonce to distinct ids) and the `msg.value` amount/fee split + dust overpayment rule. Configs points at that NatSpec.
- Minimal `PegOutNotImplemented` stubs on `FlyoverConfigurations` and `FlyoverConfigurationsMock` so the repo still compiles against the widened interface.

## Why

Same shape as the peg-in: lock the shared ABI before implementation lanes (escrow, S2 configs, SDK/LPS bindings).

## Type of Change

- [x] New feature (non-breaking change)

## Affected Areas

- [x] PegOut flow (`depositPegout`, `refundPegOut`, `refundUserPegOut`)

## Risk & Security Review

- [x] Access control and authorization paths reviewed
- [x] Reentrancy and external call paths reviewed
- [x] Storage layout / upgrade safety reviewed (if upgradeable code changed)
- [x] Time/block/confirmation assumptions reviewed
- [ ] BTC tx output/index assumptions reviewed (if pegout validation touched)
- [x] No secrets or sensitive values added

Peg-out methods on `FlyoverConfigurations` / the mock only `revert PegOutNotImplemented()` — no new storage, no deploy path changes.

## Test Plan

List what you ran locally and the outcome:

- [x] `npm run lint:sol`
- [x] `npm run lint:ts`
- [x] `npm run compile`
- [x] `npm test`

Additional notes on test coverage, edge cases, and any tests intentionally skipped:

Interface / stub-only change. Peg-in suites pass. No new escrow behavioral tests (implementation is out of scope). `test/tasks/PauseSystem.t.sol` can flake under parallel `vm.setEnv` pollution; unrelated to this PR.

## Jira Ticket

- Jira: [FLY-2541](https://rsklabs.atlassian.net/browse/FLY-2541)

## Reviewer Notes

- Treat `IPegOutEscrow` and the peg-out surface of `IFlyoverConfigurations` as frozen ABI after merge.
- Focus NatSpec decisions: id preimage includes `nonce` (`++requestCount`); amount/fee split derives `amount` from `msg.value` then `callFee = calculatePegOutFee(amount)` with dust-threshold overpayment.
- Stubs are intentional compile glue only — not the S2/S11 implementation.

### CODE REVIEW

**Review by Freshenext — APPROVED** [review]
I want to understand if there is a preference of memory vs calldata in the returns - but for the purpose of the PoC this is OK.

File: src/interfaces/IPegOutEscrow.sol [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+    /// @notice Emitted when an LP claims a REQUESTED peg-out and funds move to PegOutContract
+    /// @dev A claim is a hard commitment: there is no release / unclaim path.
+    event PegOutClaimed(address indexed lpAddress, bytes32 indexed requestHash);
+
+    /// @notice Emitted when the refund address cancels while still REQUESTED (no slash)
+    event PegOutCancelled(bytes32 indexed requestHash);
+
+    /// @notice Emitted when nobody claimed by the claim deadline and the user is refunded
+    event PegOutRefundedOnNoClaim(bytes32 indexed requestHash, address indexed refundAddress, uint256 amount);
```
Comment (Luisfc68) [inline src/interfaces/IPegOutEscrow.sol:52]: I think slither and the linter will probably ask you to index the 3 parameters

File: src/interfaces/IPegOutEscrow.sol [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+        FULFILLED,
+        REFUNDED
+    }
+
+    /// @notice Emitted when a user escrows RBTC for a peg-out (the sole commitment)
+    /// @dev Fee / deadline / confirmation snapshots live on the stored quote
+    /// ({getPegOutQuote}); this event is the LPS discovery signal only.
+    /// @param requestHash Escrow-minted request id (mapping key, LPS watch topic, BTC
+    /// OP_RETURN payload). **Id preimage (frozen):**
+    /// `requestHash = keccak256(abi.encode(chainId, address(this), nonce, msg.sender,
```
Comment (Luisfc68) [inline src/interfaces/IPegOutEscrow.sol:29]: Ok so we're actually dropping the quote then? if not maybe this could be just the quote with the LP address in 0

File: src/interfaces/IPegOutEscrow.sol [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+    /// then `amount -= amount % SAT_TO_WEI` (satoshi floor);
+    /// `callFee = calculatePegOutFee(amount)` (same satoshi-floor fee as configs).
+    /// **Overpayment rule (frozen):** `required = amount + callFee`. If
+    /// `msg.value - required` is at least the wired PegOutContract `dustThreshold`,
+    /// the excess is refunded to the refund address (legacy `depositPegOut` dust-change
+    /// semantics). Otherwise the residual is absorbed into `callFee` so escrow balance
+    /// stays attributed. `msg.value` must exceed `fixedFee` and the derived `amount`
+    /// must lie in `[minAmount, maxAmount]` or the call reverts.
+    /// @param destinationAddress User BTC payout script / address bytes
+    /// @param refundAddress Who may cancel and who receives refunds; address(0) → msg.sender
```
Comment (Luisfc68) [inline src/interfaces/IPegOutEscrow.sol:81]: Not sure about this, to avoid problems I wouldn't allow zero address in the refund and force the client to put both if they're the same

**Review by Luisfc68 — APPROVED** [review]

File: src/interfaces/IPegOutEscrow.sol (hunk 1 above)
Comment (Hakob23) [reply src/interfaces/IPegOutEscrow.sol]: Addressed in the follow-up: https://github.com/rsksmart/liquidity-bridge-contract/pull/549 — `PegOutRequested` keeps the indexed fields as reviewed there.

File: src/interfaces/IPegOutEscrow.sol (hunk 2 above)
Comment (Hakob23) [reply src/interfaces/IPegOutEscrow.sol]: Yes — request id is now `hashPegOutQuote` of the incomplete quote (`lpRskAddress = 0`). Done in https://github.com/rsksmart/liquidity-bridge-contract/pull/549.

File: src/interfaces/IPegOutEscrow.sol (hunk 3 above)
Comment (Hakob23) [reply src/interfaces/IPegOutEscrow.sol]: Agreed — zero `refundAddress` is rejected. Done in https://github.com/rsksmart/liquidity-bridge-contract/pull/549.
